src/internal_modules/roc_fec/reader.cpp

```diff
@@ -789,5 +804,24 @@ void Reader::drop_repair_packets_from_prev_blocks_() {
     }
 }
 
+void Reader::update_block_duration_(const packet::PacketPtr& ptr) {
+    if (!ptr->rtp()) {
+        return;
+    }
+    packet::stream_timestamp_diff_t block_dur = 0;
+    if (prev_block_timestamp_valid_) {
+        block_dur = packet::stream_timestamp_diff(ptr->rtp()->stream_timestamp,
+                                                  prev_block_timestamp_);
+    }
+    roc_panic_if_msg(block_dur < 0, "fec reader: negative block duration");
+    block_max_duration_ = std::max(block_max_duration_, block_dur);
```

Comment:
Shouldn't we compute maximum in LatencyTuner instead of fec::Reader? And report only immediate value from fec::Reader?

Why:

* Most likely we want running maximum instead of total maxium. Maybe at least in later versions. But this kind of details (running window, etc) should belong to LatencyTuner?

* This logic (running max) would probably be the same for sender & receiver.

Reply:
With the manner the metrics are passed into tuner, it would be difficult to be sure that it has updates of fec block duration on every new block arrived. And this could potentially lead to  wrong estimation of maximum.

Reply:
Fair enough.

Reply:
Hm, given it one more thought, this is probably not the case.

While latency tuner updates FE with some big period, like 200ms, latency monitor and feedback monitor are invoked *every frame*.

And, while user is allowed to read and write big frames, top-level pipeline code will anyway split them in smaller frames (to ensure that we don't block in processing for long time and that frames can fit max size of pool buffer).

So in practice all pipeline components are invoked with small frames (e.g. 1ms), much lower than FEC block size, and I think in our design (in general) we can assume that it's always true, because it simplifies many things.

For example, we already do the same with incoming queue - we're sampling it's size every frame. I think we in general can sample anything per frame.

So I guess you can just sample FEC block duration in LatencyMonitor::read?

Reply:
It is not an easy task to reproduce this reasoning based on source code only, so my concern could potentially be raised during some debug session and it would be difficult to confirm or deny, so I'd rather keep it as it is

Reply:
Makes sense.